enhancements/builds/multi-arch-builds.md

```diff
@@ -0,0 +1,456 @@
+>   stability, availability, performance and security.
+> - Describe which OCP teams are likely to be called upon in case of escalation with one of the failure modes
+>   and add them as reviewers to this enhancement.
+
+* If all builders for any one architecture become unavailable, all builds will ultimately fail.
+
+#### Support Procedures
+
+> Describe how to
+> - detect the failure modes in a support situation, describe possible symptoms (events, metrics,
+>   alerts, which log output in which component)
+>
+>   Examples:
+>   - If the webhook is not running, kube-apiserver logs will show errors like "failed to call admission webhook xyz".
+>   - Operator X will degrade with message "Failed to launch webhook server" and reason "WehhookServerFailed".
+>   - The metric `webhook_admission_duration_seconds("openpolicyagent-admission", "mutating", "put", "false")`
+>     will show >1s latency and alert `WebhookAdmissionLatencyHigh` will fire.
+>
+> - disable the API extension (e.g. remove MutatingWebhookConfiguration `xyz`, remove APIService `foo`)
+>
+>   - What consequences does it have on the cluster health?
+>
+>     Examples:
+>     - Garbage collection in kube-controller-manager will stop working.
+>     - Quota will be wrongly computed.
+>     - Disabling/removing the CRD is not possible without removing the CR instances. Customer will lose data.
+>       Disabling the conversion webhook will break garbage collection.
+>
+>   - What consequences does it have on existing, running workloads?
+>
+>     Examples:
+>     - New namespaces won't get the finalizer "xyz" and hence might leak resource X
+>       when deleted.
+>     - SDN pod-to-pod routing will stop updating, potentially breaking pod-to-pod
+>       communication after some minutes.
+>
+>   - What consequences does it have for newly created workloads?
+>
+>     Examples:
+>     - New pods in namespace with Istio support will not get sidecars injected, breaking
+>       their networking.
+>
+> - Does functionality fail gracefully and will work resume when re-enabled without risking
+>   consistency?
+>
+>   Examples:
+>   - The mutating admission webhook "xyz" has FailPolicy=Ignore and hence
+>     will not block the creation or updates on objects when it fails. When the
+>     webhook comes back online, there is a controller reconciling all objects, applying
+>     labels that were not applied during admission webhook downtime.
+>   - Namespaces deletion will not delete all objects in etcd, leading to zombie
+>     objects when another namespace with the same name is created.
+
+## Implementation History
+
+## Alternatives
+
+* Have the build controller generate N arch-specific Build objects for a BuildConfig, and a final custom-strategy Build which is told to monitor the N Builds and stitch their outputs together into a manifest list.
+  * `oc start-build` would no longer be sufficient to retry a failed build.
```

Comment:
is this the only reason to reject this alternative?  i'd like to see some more discussion of the pros/cons of this approach, because the idea of having a build object that represents each actual build seems nice to me in a lot of ways. 

and we could always augment oc start-build to handle multiarch builds differently.